changelogs/fragments/fix-snmp-server-idempotency.yml

```diff
@@ -0,0 +1,3 @@
+---
+bugfixes:
+  - ios_snmp_server - Fixed the idempotency issue in SNMP configuration management caused by in-place modifications in compare lists method, using key tracking to ensure accurate updates without unintended overwrites
```

Comment:
This still has some technical elements in the changelog, we have to give an apt description about the problem, for example `Fixes an issue where enabling the read-only (ro) attribute in communities wasn't idempotent.`, This focuses on what was fixed rather than how it was fixed in the code.

Reply:
sure will change it